### PR TITLE
Kanban Node (Task) Modal

### PR DESCRIPTION
**Pull request by schromya**
+ Added functionality so that when you click on a node in the Kanban board, the node modal shows up.
+ Fixed bug with new task/node button where it causes a route error when it's clicked when the node modal is pulled up both on graph and kanban.
+ Fixed bug where Graph/Kanban Button had incorrect name when certain modals where pulled up.

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def taskView(request, projectID, taskID):
 
     if "kanban" in request.path:
         baseTemplate = "kanban.html"
-        submitLink = f"/project/{projectID}/kanban/task/{taskID}"
+        submitLink = f"/project/{projectID}/kanban/task/{taskID}/"
         exitLink = f"/project/{projectID}/kanban"
         deleteLink = f"/project/{projectID}/kanban"
     else:
```
Comment (crazykyler005) [inline app/views.py]: I would change this so there is defaults set for these variables instead
```suggestion
    baseTemplate = "kanban.html"
    submitLink = f"/project/{projectID}/kanban/task/{taskID}"
    submitLink = f"/project/{projectID}/kanban/task/{taskID}/"
    exitLink = f"/project/{projectID}/kanban"
    deleteLink = f"/project/{projectID}/kanban"
    
    if "kanban" not in request.path:
    	baseTemplate = "graph.html"
        submitLink = f"/project/{projectID}/graph/task/{taskID}/"
        exitLink = f"/project/{projectID}/graph"
        deleteLink = f"/project/{projectID}/graph"
```

File: app/templates/project.html [hunk 2]
```diff
@@ -102,20 +102,19 @@
...
     document.getElementById('formatButton').href = '/project/{{projectID}}/graph';
     document.getElementById('formatText').textContent = 'Graph';
+    document.getElementById('newTaskButton').href = href="/project/{{projectID}}/kanban/create-task/"
+
+  } else {
+    document.getElementById('formatButton').href = '/project/{{projectID}}/kanban';
+    document.getElementById('formatText').textContent = 'Kanban';
+    document.getElementById('newTaskButton').href = href="/project/{{projectID}}/graph/create-task/"
+
   }
```
Comment (crazykyler005) [inline app/templates/project.html:118]: same here
```suggestion
  document.getElementById('formatButton').href = '/project/{{projectID}}/graph';
  document.getElementById('formatText').textContent = 'Graph';
  document.getElementById('newTaskButton').href = href="/project/{{projectID}}/kanban/create-task/"

  if not pathSegments.includes('kanban') {
    document.getElementById('formatButton').href = '/project/{{projectID}}/kanban';
    document.getElementById('formatText').textContent = 'Kanban';
    document.getElementById('newTaskButton').href = href="/project/{{projectID}}/graph/create-task/"

  }
```

File: app/templates/kanban.html
Comment (erwilliams64) [inline app/templates/kanban.html:1]: Is this extensible for when new statuses are added?

File: app/templates/kanban.html [hunk 3]
```diff
@@ -52,42 +52,26 @@
...
-        listLink.textContent = task.name
-        listLink.draggable = 'true'
-        listLink.setAttribute('ondragstart', 'drag(event)')
-        listLink.className = 'block rounded-lg my-2 p-1 hover:bg-neutral-500 bg-neutral-600 '
-        listLink.setAttribute('priority', task.priority)
-        doneListElement.appendChild(listLink)
-      })
+
+
+      for (status of ['Todo', 'In-Progress', 'Done']) {
```
Comment (erwilliams64) [inline app/templates/kanban.html:57]: Is this extensible for when new statuses are added?

Comment (schromya) [reply app/templates/kanban.html:57]: It should be. Instead of the "hardcoded" status list, you would give it the list of custom statuses.

**Review by crazykyler005 — APPROVED** [review]
Eh, good enough.

**Pull request merged**